dotCMS/src/main/java/com/dotcms/ema/proxy/ProxyTool.java

```diff
@@ -0,0 +1,1214 @@
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.apache.http.client.CredentialsProvider;
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpDelete;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpHead;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpPut;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.message.BasicNameValuePair;
+import org.apache.http.util.EntityUtils;
+
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.UtilMethods;
+
+/**
+ * This viewtool provides a method to make a simple post request and get the value.
+ * 
+ * @author Aquent, LLC. (cfalzone@aquent.com)
+ */
+public class ProxyTool {
+
+
+
+    private static final String DEFAULT_CONTENT_TYPE = "application/x-www-form-urlencoded";
+
+    public static final int ERR_CODE_UNKNOWN_ERR = 888;
+    public static final int ERR_CODE_UNIMPLEMENTED_METHOD = 777;
+
+    public static final String METHOD_PUT = "PUT";
+    public static final String METHOD_POST = "POST";
+    public static final String METHOD_GET = "GET";
+    public static final String METHOD_HEAD = "HEAD";
+    public static final String METHOD_DELETE = "DELETE";
+
+
+
+    /**
+     * Get a credentials object to send to authenticated requests.
+     *
+     * @param user - The username
+     * @param pass - The password
+     * @return A Credentials object that can be used in authenticated requests.
+     */
+    public Credentials createCreds(String user, String pass) {
+        return new UsernamePasswordCredentials(user, pass);
+    }
+
+    /**
+     * This is used to send a single string payload to a url. Can be used to send JSON or XML to a url.
+     * Only Supports POST and PUT methods.
+     *
+     * @param url - The URL
+     * @param data - The String Data
+     * @param method - The Method (POST/PUT)
+     * @param contentType - The content type for this request
+     * @param creds - A credentials object for authenticated requests.
+     * @return A ProxyResponse Object
+     */
+    public ProxyResponse sendStringData(String url, String data, String method, String contentType, Credentials creds) {
+        Logger.debug(this, "sendStringData called with url=" + url + ", data=" + data + ", and method=" + method + ", contentType = "
+                + contentType + ", creds=" + UtilMethods.isSet(creds));
+
+        StringEntity entity;
+        try {
+            entity = new StringEntity(data, ContentType.create(contentType, "UTF-8"));
+        } catch (Exception e) {
```

Comment:
Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5922984)